src/overlays/actors/ovl_Oceff_Storm/z_oceff_storm.c

```diff
@@ -33,22 +31,183 @@ const ActorInit Oceff_Storm_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981928.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981B48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981BB8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Draw.s")
+void OceffStorm_SetupAction(OceffStorm* this, OceffStormActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+s32 func_8098176C(PlayState* play) {
+    s32 phi_v1 = false;
+
+    switch (play->sceneNum) {
+        case SCENE_13HUBUKINOMITI:
+        case SCENE_11GORONNOSATO:
+        case SCENE_10YUKIYAMANOMURA:
+        case SCENE_14YUKIDAMANOMITI:
+        case SCENE_12HAKUGINMAE:
+        case SCENE_17SETUGEN:
+        case SCENE_GORONRACE:
+            if (gSaveContext.sceneSetupIndex == 0) {
+                phi_v1 = true;
+            }
+            break;
+        case SCENE_10YUKIYAMANOMURA2:
+            if (gSaveContext.sceneSetupIndex == 1) {
+                phi_v1 = true;
+            }
+            break;
+    }
+    return phi_v1;
+}
+
+void OceffStorm_Init(Actor* thisx, PlayState* play) {
+    s32 pad[2];
+    Player* player = GET_PLAYER(play);
+    OceffStorm* this = THIS;
+
+    OceffStorm_SetupAction(this, OceffStorm_DefaultAction);
+
+    this->posYOffAdd = 0;
+    this->counter = 0;
+    this->primColorAlpha = 0;
+    this->vtxAlpha = 0;
+    this->actor.scale.y = 0.0f;
+    this->actor.scale.z = 0.0f;
+    this->actor.scale.x = 0.0f;
+    this->posYOff = this->posYOffAdd;
+
+    if (this->actor.params == 1) {
+        OceffStorm_SetupAction(this, func_80981B48);
+        this->actor.draw = OceffStorm_Draw2;
+    } else {
+        this->actor.world.pos.y = player->actor.world.pos.y;
+        this->actor.world.pos.x = player->bodyPartsPos[0].x;
+        this->actor.world.pos.z = player->bodyPartsPos[0].z;
+        gSaveContext.jinxTimer = 0;
+        if ((play->interfaceCtx.restrictions.songOfStorms == 0) && !func_8098176C(play)) {
+            Actor_Spawn(&play->actorCtx, play, ACTOR_EN_OKARINA_EFFECT, this->actor.world.pos.x,
+                        this->actor.world.pos.y - 30.0f, this->actor.world.pos.z, 0, 0, 0, 1);
+        }
+    }
+}
+
+void OceffStorm_Destroy(Actor* thisx, PlayState* play) {
+    OceffStorm* this = THIS;
+
+    Magic_Reset(play);
+}
+
+void OceffStorm_DefaultAction(OceffStorm* this, PlayState* play) {
+    f32 cylinderScale;
+
+    switch (gSaveContext.save.playerForm) {
+        default:
+            cylinderScale = 1.0f;
+            break;
+        case PLAYER_FORM_DEKU:
+            cylinderScale = 1.3f;
+            break;
+        case PLAYER_FORM_ZORA:
+            cylinderScale = 1.2f;
+            break;
+        case PLAYER_FORM_GORON:
+            cylinderScale = 2.0f;
+            break;
```

Comment:
```suggestion
            break;

        case PLAYER_FORM_DEKU:
            cylinderScale = 1.3f;
            break;

        case PLAYER_FORM_ZORA:
            cylinderScale = 1.2f;
            break;

        case PLAYER_FORM_GORON:
            cylinderScale = 2.0f;
            break;
```
Makes it clearer that no fallthrough is present.